### PR TITLE
Add log message for __precompile__(false)

### PR DESCRIPTION
**Pull request by NHDaly**
Add a log message that we're bailing on precompilation because we encountered `__precompile__(false)`.

Also states that we're importing the package the "normal way" to help explain why this can still be slow.

Opened in response to https://github.com/JuliaLang/julia/issues/29467.

### CODE REVIEW

**Comment by NHDaly** [comment]
This isn't a perfect solution; ideally we would never print `Precompiling $pkg` at all, but it's not possible to know whether we'll end up finding a `__precompile__(false)` statement somewhere in the package until we've already finished precompiling!

So I think this is a decent workaround -- just letting the user know what's happening: 1. we've bailed on precompilation, and 2. now we're gonna import the package the regular way.


Bikeshedding on the exact wording of this log message is requested. :) Thanks!

**Comment by NHDaly** [comment]
@KristofferC can i bug you for a review of this? I'm just going back to clean up old PRs. :)

It's a small logging change, but i think a good usability improvement!

**Review by KristofferC — APPROVED** [review]
Seems better than the status quo so if you have verified that it works correctly, LGTM.

**Comment by NHDaly** [comment]
> so if you have verified that it works correctly, LGTM.

Yes, it seems to work locally.

```julia
pkg> add StatPlots#v0.8.1
...

julia> using StatPlots
[ Info: Precompiling StatPlots [60ddc479-9b66-56df-82fc-76a74619b69c]
[ Info: Skipping precompilation since __precompile__(false). Importing StatPlots [60ddc479-9b66-56df-82fc-76a74619b69c].

julia> using StatPlots

julia>
```
And then the second time after restarting julia:
```julia
julia> using StatPlots
[ Info: Recompiling stale cache file /Users/nathan.daly/.julia/compiled/v1.2/StatPlots/iAmZm.ji for StatPlots [60ddc479-9b66-56df-82fc-76a74619b69c]
[ Info: Skipping precompilation since __precompile__(false). Importing StatPlots [60ddc479-9b66-56df-82fc-76a74619b69c].
```

It's better than nothing, and at least it gives you an understanding of why it's taking so long! :)

Thanks for the review.

File: base/loading.jl [hunk 1]
```diff
@@ -959,7 +959,10 @@ function _require(pkg::PkgId)
                 # or if the require search declared it was pre-compiled before (and therefore is expected to still be pre-compilable)
                 cachefile = compilecache(pkg, path)
                 if isa(cachefile, Exception)
-                    if !precompilableerror(cachefile)
+                    if precompilableerror(cachefile)
+                         verbosity = isinteractive() ? CoreLogging.Info : CoreLogging.Debug
```
[the remark this answers is not in the document]
Comment (NHDaly) [reply base/loading.jl]: Yes, thanks, good call! Fixed it in a commit through the github interface.

**Comment by NHDaly** [comment]
Please take another look: I merged in the latest master, and the build now passes. :)

I think this is good to merge, and would be a useful change to prevent further confusion. :)

**Comment by KristofferC** [comment]
Thanks for the work here!

**Pull request merged**